webapp/src/store/Store.ts

```diff
@@ -1,7 +1,59 @@
+import fs from 'fs/promises';
+
 import { ProjectStore } from '@/store/ProjectStore';
+import MessageAdapterFactory from '@/utils/adapters/MessageAdapterFactory';
+import YamlTranslationAdapter from '@/utils/adapters/YamlTranslationAdapter';
+import { LyraProjectConfig } from '@/utils/lyraConfig';
+import { StoreData } from './types';
+
+const FILE_PATH = './store.json';
 
 export class Store {
   private data = new Map<string, ProjectStore>();
+  private initialState: Record<string, StoreData | undefined> = {};
+
+  public static async getProjectStore(
+    lyraProjectConfig: LyraProjectConfig,
+  ): Promise<ProjectStore> {
+    if (!globalThis.store) {
+      globalThis.store = new Store();
+      await globalThis.store.loadFromDisk();
+    }
```

Comment:
While the first call to `getProjectStore` is awaiting `loadFromDisk`, concurrent calls will find `globalThis.store` true-like and continue working with an empty object as `initialState`.

These concurrent calls will continue initializing ProjectStore using the empty `initialState`. When `loadFromDisk` resolves, the initial call will find ProjectStore already initialized and the loaded state will never be used.

`accessProjects` uses `getProjectStore` concurrently for all projects so the risk of this bug is quite high. We will find a resolution for it in this PR before merging.